tests/starknet/scarb.rs

```diff
@@ -0,0 +1,57 @@
+use anyhow::{anyhow, Error};
+use scarb::{
+    core::{Config, PackageId, PackageName, SourceId, TargetKind},
+    ops::{self, CompileOpts, FeaturesOpts, FeaturesSelector},
+};
+use semver::Version;
+use std::{fs::canonicalize, path::PathBuf};
+
+#[allow(dead_code)]
+pub struct Compiler {
+    toml: PathBuf,
+    opts: CompileOpts,
+    packages: Vec<PackageId>,
+}
+
+#[allow(dead_code)]
+impl Compiler {
+    pub fn new(toml: &str) -> Result<Self, Error> {
+        let toml_absolute = canonicalize(PathBuf::from(toml))?;
+        let opts = CompileOpts {
+            include_target_kinds: vec![],
+            exclude_target_kinds: vec![TargetKind::new("test")],
+            include_target_names: vec![],
+            features: FeaturesOpts {
+                features: FeaturesSelector::Features(vec![]),
+                no_default_features: false,
+            },
+        };
+        let packages = vec![PackageId::new(
+            PackageName::new("account"),
+            Version::new(0, 1, 0),
+            SourceId::for_path(toml_absolute.to_str().unwrap().into())?,
+        )];
+
+        Ok(Compiler { toml: toml_absolute, opts, packages })
+    }
+
+    pub async fn compile(self) -> Result<(), Error> {
```

Comment:
Why not just make the method blocking and let client code decide whether it needs to be run with `spawn_blocking` or not?

Remember, "premature optimization is the root of all evil."